modules/system/rolls/cast-test.js

```diff
@@ -182,9 +223,19 @@ export default class CastTest extends TestWFRP {
 
 
   postTest() {
-    // Find ingredient being used, if any
-    if (this.hasIngredient && this.item.ingredient.quantity.value > 0 && !this.context.edited && !this.context.reroll)
-      this.item.ingredient.update({ "data.quantity.value": this.item.ingredient.quantity.value - 1 })
+    //@/HOUSE
+    let homeBrewIngredient = (this.preData.unofficialGrimoire && this.preData.ingredientMode != 'none') || !this.preData.unofficialGrimoire;
+    if (homeBrewIngredient) {
+      if (this.hasIngredient && this.item.ingredient.quantity.value > 0 && !this.context.edited && !this.context.reroll) {
+        this.item.ingredient.update({ "data.quantity.value": this.item.ingredient.quantity.value - 1 })
+        ChatMessage.create({ speaker: this.data.context.speaker, content: game.i18n.localize("ConsumedIngredient") })
+      }
+    //@/HOUSE
+    } else {
+      // Find ingredient being used, if any
+      if (this.hasIngredient && this.item.ingredient.quantity.value > 0 && !this.context.edited && !this.context.reroll)
+        this.item.ingredient.update({ "data.quantity.value": this.item.ingredient.quantity.value - 1 })
+    }
```

Comment:
Can you add... you guessed it...   `game.wfrp4e.utility.logHomebrew("unofficialgrimoire")`

Reply:
done